utils/manifest-utils/src/util.rs

```diff
@@ -15,6 +27,271 @@ pub enum DecodingError {
+        }
+        return Ok(KeystoreLoadResult::Loaded { loaded, skipped });
+    }
+
+    pub fn remove_certs(self, ids: &HashSet<String>) -> anyhow::Result<KeystoreRemoveResult> {
+        if ids.difference(&self.ids).eq(ids) {
+            return Ok(KeystoreRemoveResult::NothingToRemove);
+        }
+        let mut removed = HashMap::new();
+
+        let cert_dir_entries: Vec<Result<DirEntry, std::io::Error>> =
+            std::fs::read_dir(self.cert_dir.clone())?.collect();
+        for dir_entry in cert_dir_entries {
+            let cert_file = dir_entry?;
+            let cert_file = cert_file.path();
+            let certs = parse_cert_file(&cert_file)?;
+            if certs.is_empty() {
+                // No certificates in parsed file.
+                continue;
+            }
+            let mut ids_cert = certs.into_iter().fold(HashMap::new(), |mut certs, cert| {
+                if let Ok(id) = cert_to_id(&cert) {
+                    certs.insert(id, cert);
+                }
+                certs
+            });
+            let mut split_and_skip = false;
+            for id in ids.iter() {
+                if let Some(cert) = ids_cert.remove(id) {
+                    removed.insert(id, cert);
+                    split_and_skip = true;
+                }
+            }
+            if split_and_skip {
+                let file_stem = get_file_stem(&cert_file).expect("Cannot get file name stem");
+                let dot_extension = get_file_extension(&cert_file)
+                    .map_or_else(|| String::from(""), |ex| format!(".{}", ex));
+                for (id, cert) in ids_cert {
+                    let cert = cert.to_pem()?;
+                    let mut file_path = self.cert_dir.clone();
+                    let filename = format!("{}.{}{}", file_stem, id, dot_extension);
+                    file_path.push(filename);
+                    fs::write(file_path, cert)?;
+                }
+                fs::remove_file(cert_file)?;
+            }
+        }
+
+        let removed: Vec<X509> = removed.into_values().collect();
+        Ok(KeystoreRemoveResult::Removed { removed })
+    }
+
+    /// Loads keychain file to `cert_dir`
+    fn load_as_keychain_file(&self, cert_path: &PathBuf) -> anyhow::Result<()> {
+        let file_name = get_file_name(cert_path)
+            .ok_or_else(|| format!("Cannot get filename of {:?}", cert_path))
+            .map_err(|err| anyhow::anyhow!(err))?;
+        let mut new_cert_path = self.cert_dir.clone();
+        new_cert_path.push(file_name);
+        if new_cert_path.exists() {
+            let file_stem = get_file_stem(&new_cert_path).expect("Has to have stem");
+            let dot_extension = get_file_extension(&new_cert_path)
+                .map(|ex| format!(".{}", ex))
+                .unwrap_or(String::from(""));
+            for i in 0..u32::MAX {
+                let numbered_filename = format!("{0}.{1}{2}", file_stem, i, dot_extension);
```

Comment:
That should probably work and look better:
```suggestion
                let numbered_filename = format!("{file_stem}.{i}{dot_extension}");
```

Reply:
Oslo consider using it in other places where fmt is used